Add Spring Boot migration course lab guides (#397)
moderneinc/moderne-docs: docs/hands-on-learning/spring-boot-migration/module-5-querydsl-and-wave-upgrades.md

:::warning
You can't use recipe builder in this case since you are referring to a custom recipe that is not available in the platform.
:::

2. Install this custom recipe locally with the CLI:

```bash
mod config recipes yaml install CustomUpgradeSpringBoot_4_0.yml
```

## Exercise 5-3: Upgrade wave 0

### Goals for this exercise

* Apply the composite recipe to wave 0
* Build and release wave 0

### Steps

Now that the custom upgrade recipe is ready, it's time to upgrade each wave! 

1. Start by upgrading Wave 0 using the following commands. 

```bash
mod run $WORKSPACE/Wave0 --recipe org.openrewrite.recipe.querydsl.CustomUpgradeSpringBoot_4_0
mod git apply $WORKSPACE/Wave0 --last-recipe-run
```

:::note
Notice you are referencing the `Wave0` directory for each of the `mod` commands. This is taking advantage of the organizational directory structure to target only the wave you want to apply changes to so you can do each wave individually.
:::

2. Now that the changes have been applied, verify the build and then release:

```bash
$WORKSHOP/build.sh 0
$WORKSHOP/release.sh 0
```

> <!-- TODO: Could we potentially skip the build step here if we're confident that it works? The release does the build as well, no? -->

Reviewer: yes think so, at least when we demo it.

This isn't surprising since you saw Wave 0 successfully build earlier. It was Wave 1 that failed last time, so the real test will be upgrading Wave 1.

## Exercise 5-4: Upgrade wave 1

### Goals for this exercise

* Apply the composite recipe to wave 1
* Build and release wave 1

### Steps

1. Once again, run the recipe, but this time, targeting the `Wave1` directory:

```bash
mod run $WORKSPACE/Wave1 --recipe org.openrewrite.recipe.querydsl.CustomUpgradeSpringBoot_4_0
mod git apply $WORKSPACE/Wave1 --last-recipe-run
```

2. This time, build Wave 1 to make sure there are no errors, then release it:

```bash
$WORKSHOP/build.sh 1
$WORKSHOP/release.sh 1
```

## Exercise 5-5: Refresh DevCenter and continue waves

### Goals for this exercise

* Refresh DevCenter and confirm progress
* Continue running wave upgrades sequentially

### Steps

> At the end of Module 1, you used an organizational DevCenter to see the status of the migration across repositories. Now that you've successfully upgrade a handful of repositories, you can generate a new DevCenter dashboard locally to compare and track your progress.

Reviewer 2: ```suggestion
At the end of Module 1, you used an organizational DevCenter to see the status of the migration across repositories. Now that you've successfully upgraded a handful of repositories, you can generate a new DevCenter dashboard locally to compare and track your progress.
```

    > Run mod git apply /Users/somebody/workspaces/migration_workshop --recipe-run 20260115175824-jLfJP to apply the changes
    > Run mod log runs add /Users/somebody/workspaces/migration_workshop logs.zip --last-recipe-run to aggregate run logs
    > Examine run telemetry

MOD SUCCEEDED in 9s
   ▛▀▀▚▖  ▗▄▟▜
   ▌   ▜▄▟▀  ▐
   ▛▀▀█▀▛▀▀▀▀▜
   ▌▟▀  ▛▀▀▀▀▜
   ▀▀▀▀▀▀▀▀▀▀▀
Moderne CLI 3.54.5

⏺ Reading organization

Found 1 organization containing 11 repositories (1s)
Found recipe run 20260115175824-jLfJP

⏺ Building CSV for upgrades and migrations

▶ Adhoc repository group
    ✓ Data table produced
Done (1s)

⏺ Building CSV for security issues

▶ Adhoc repository group
    ✓ Data table produced
Done (1s)

⏺ Generate DevCenter dashboard

Full DevCenter log

▶ Adhoc repository group
    ✓ Generated dashboard
Done (3s)

DevCenter generated

MOD SUCCEEDED in 4s
```

</details>

3. Open the generated HTML file from the `mod devcenter` output by using ctrl-click (Windows) or cmd-click (Mac) on the `Generated dashboard` link in the output.

4. You should notice:

  * Java 17 is applied across the org
  * Wave 0 and Wave 1 now show Spring Boot 4.0
  * Remaining waves are still pending

> 5. You can now continue upgrading waves until the entire portfolio is on Spring Boot 4. Use the commands as in exercise 5-4, each time targeting the next wave:

Reviewer 2: ```suggestion
5. You can now continue upgrading waves until every repository is on Spring Boot 4. Use the commands as in exercise 5-4, each time targeting the next wave:
```

```bash
mod run $WORKSPACE/Wave2 --recipe org.openrewrite.recipe.querydsl.CustomUpgradeSpringBoot_4_0
mod git apply $WORKSPACE/Wave2 --last-recipe-run
$WORKSHOP/build.sh 2
$WORKSHOP/release.sh 2

# ...and so on
```

> ### Takeaways

Reviewer 2: ```suggestion
## Takeaways
```
